template/koa-rest-api/src/api/jobs/postJob.test.ts

```diff
@@ -27,11 +27,17 @@ describe('postJobHandler', () => {
       .expect(422)
       .expect(({ text }) =>
         expect(text).toMatchInlineSnapshot(`
-          "Validation failed:
-          {
-            "hirer": "Expected { id: string; }, but was missing"
-          }.
-          Object should match { hirer: { id: string; }; }"
+          "[
```

Comment:
I think we should discourage returning zod errors and return a [JsonResponse](https://github.com/seek-oss/koala/blob/0852e01d25ba071dabef559df8cb5b77dff8636e/src/errorMiddleware/README.md#jsonresponse)  instead:
```ts
{
  "message": "Input validation failed",
  "invalidFields": {
    "/hirer": "Required"
  }
}
```
[Here](https://github.com/SEEK-Jobs/hirer-accounts-api/blob/master/src/api/adapters/parseInvalidFieldsFromError.ts) is a function that converts zod errors to invalid fields

Reply:
Good catch, @brian-captain-crosby.
I feel like that `JsonResponse` is more readable compared to the Zod errors.
I have replaced Zod errors with `JsonResponse` in `koa-rest-api`.